### DISCUSSION TITLE
The newline character in PyMuPDF

### DISCUSSION THREAD

**Discussion by dummifiedme** [pymupdf/PyMuPDF#830] [Looking for help] [answered]
I noticed that there is a `\r` char at the end of a new line in comments upon extraction
Is it due to PyMuPDF or the PDF viewer which was used to mark the pdf. And, can that system of newline changed?

**Comment by JorjMcKie** [comment]
PyMuPDF adds no characters when extracting text. That `\r` (backspace) is part of what the creator did.
Text extraction offers flags to influence its outut see [here](https://pymupdf.readthedocs.io/en/latest/vars.html#preserve-text-flags). Among other things, white spaces can be converted to blanks.

**Comment by dummifiedme** [comment, marked as answer]
Okay, I suspected the same. I will look into the link.
Thanks a lot!
